platform/security/sts/security-sts-realm/src/main/java/ddf/security/realm/sts/AbstractStsRealm.java

```diff
@@ -237,24 +244,32 @@ protected SecurityToken requestSecurityToken(Object authToken) {
      * @param securityToken The token being renewed.
      * @return security token (SAML assertion)
      */
-    protected SecurityToken renewSecurityToken(SecurityToken securityToken) {
+    protected SecurityToken renewSecurityToken(final SecurityToken securityToken) {
         SecurityToken token = null;
         String stsAddress = getAddress();
 
         try {
             LOGGER.debug("Renewing security token from STS at: {}.", stsAddress);
 
             if (securityToken != null) {
-                LOGGER.debug("Telling the STS to renew a security token on behalf of the auth token");
-                STSClient stsClient = configureStsClient();
-
-                stsClient.setWsdlLocation(stsAddress);
-                stsClient.setTokenType(getAssertionType());
-                stsClient.setKeyType(getKeyType());
-                stsClient.setKeySize(Integer.parseInt(getKeySize()));
-                stsClient.setAllowRenewing(true);
-                token = stsClient.renewSecurityToken(securityToken);
-                LOGGER.debug("Finished renewing security token.");
+                synchronized (securityToken.getToken()) {
+                    SecurityToken cachedToken = cache.getIfPresent(securityToken.getToken());
```

Comment:
I changed to `expireAfterAccess`. Ideally this could only occur if the system was under a huge amount of strain. You'd need X number of requests to all come in at the same time, the first one gets through and then the remaining X-1 requests come in staggered by some large amount of time. Any other requests coming in to the system would already get the new token that the first request returned and wouldn't hit this logic anyways. This way it provides for about a minute between each of the requests before finally tossing it out of the cache.